src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,64 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import React from 'react';
+import { StatusOfFilter } from '../TodoList';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+type Props = {
+  status: StatusOfFilter;
+  handleChange: (value: StatusOfFilter) => void;
+  query: string;
+  setQuery: (arg: string) => void;
+};
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+export const TodoFilter: React.FC<Props> = (
+  {
+    status,
+    handleChange,
+    query,
+    setQuery,
+  },
+) => {
+  return (
+    <form className="field has-addons">
```

Comment:
It's a good practice to use the 'onSubmit' event of the form to handle possible form submissions. You can use 'event.preventDefault()' to prevent the default submit behavior. 📝